src/v1/controllers/auth.controller.js

```diff
@@ -0,0 +1,64 @@
+const User = require('../models/user.model')
+const jwt = require('jsonwebtoken')
+const bcrypt = require('bcrypt')
+const catchAsync = require("../../utils/catchAsync");
+require('dotenv').config()
+
+const register =  catchAsync( async (req, res) => {
+    try {
+        const user = User.create(req.body)
+        const token = jwt.sign({id: user.id}, process.env.JWT_SECRET, {expiresIn: '1h'})
+        res.json({
+            token,
+            user: {
+                id: user.id,
+                email: user.email,
+                role: user.role
```

Comment:
1. What are the roles? The roles should be defined and validated.
2. Is it a good idea to allow anyone to signup with any role? What if someone is able to register with a privileged role during the sign up?